mlflow/tracing/fluent.py

```diff
@@ -211,7 +214,7 @@ def __exit__(self, exc_type, exc_value, traceback):
             # of start_span and OTel's use_span can execute).
             if exc_type is not None:
                 self.coro.throw(exc_type, exc_value, traceback)
-            self.coro.close()
+            self.coro.close()  # This triggers GeneratorExit
```

Comment:
Is this line really necessary?

Reply:
I've never used `close`. Reading https://docs.python.org/3.13/reference/expressions.html#generator.close now.

Reply:
When this line is called, we're pausing at `yield result`.